src/com/atlas/controller/TitleController.java

```diff
@@ -18,18 +23,54 @@
 public class TitleController {
 
 	@Autowired
+	@Qualifier("titleServiceImpl")
 	private TitleService titleService;
 
 	@RequestMapping(method = RequestMethod.GET, produces=MediaType.APPLICATION_JSON_VALUE)
 	public List<Title> getAllTitles() {
 		return this.titleService.listTitles();
 	}
+	
+	@RequestMapping(value="{id}",method = RequestMethod.GET, produces=MediaType.APPLICATION_JSON_VALUE)
+	public Title getTitleById(@PathVariable("id") int id) throws MovieNotFound{
+		return this.titleService.getTitleById(id);
+	}
 
 	@RequestMapping(method = RequestMethod.POST, produces=MediaType.APPLICATION_JSON_VALUE,
 			consumes=MediaType.APPLICATION_JSON_VALUE)
-	public Title addTitle(@RequestBody Title t) {
-		this.titleService.addTitle(t);
-		return t;
+	public Title addTitle(@RequestBody Title t) throws MovieBadRequest{
```

Comment:
I am returning this object as we need to confirm on UI side too that we have this entire object in list. if we just return HttpStatus, we can't show them in grid instantly after add.